src/imagings/ViewImagings.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Table } from '@hospitalrun/components'
+import format from 'date-fns/format'
+import React, { useState, useEffect, useCallback } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+
+import { useButtonToolbarSetter } from '../page-header/button-toolbar/ButtonBarProvider'
+import useTitle from '../page-header/title/useTitle'
+import useTranslator from '../shared/hooks/useTranslator'
+import Permissions from '../shared/model/Permissions'
+import { RootState } from '../shared/store'
+import { extractUsername } from '../shared/util/extractUsername'
+import { searchImagings } from './imagings-slice'
+
+type ImagingFilter = 'requested' | 'completed' | 'canceled' | 'all'
+
+const ViewImagings = () => {
+  const { t } = useTranslator()
+  const history = useHistory()
+  const setButtons = useButtonToolbarSetter()
+  useTitle(t('imagings.label'))
+
+  const { permissions } = useSelector((state: RootState) => state.user)
+  const dispatch = useDispatch()
+  const { imagings } = useSelector((state: RootState) => state.imagings)
+  const [searchFilter, setSearchFilter] = useState<ImagingFilter>('all')
+
+  const getButtons = useCallback(() => {
+    const buttons: React.ReactNode[] = []
+
+    if (permissions.includes(Permissions.RequestImaging)) {
+      buttons.push(
+        <Button
+          icon="add"
+          onClick={() => history.push('/imagings/new')}
+          outlined
+          color="success"
+          key="imaging.requests.new"
+        >
+          {t('imagings.requests.new')}
+        </Button>,
+      )
+    }
+
+    return buttons
+  }, [permissions, history, t])
+
+  useEffect(() => {
+    setSearchFilter('all' as ImagingFilter)
+  }, [])
```

Comment:
is this `useEffect` needed? it seems like it might not be since we initialize the state to be `all`

Reply:
This method is to display all the imaging with any status (`requested`, `canceled`, and `completed`). And this method can be used for `searching` imaging with specific status in the future.